src/clouds.cpp

```diff
@@ -345,15 +351,16 @@ void Clouds::render()
 
 void Clouds::step(float dtime)
 {
-	m_time += dtime;
+	m_origin = m_origin + dtime * BS * m_speed;
 }
 
-void Clouds::update(v2f camera_p, video::SColorf color)
+void Clouds::update(v2f camera_p, video::SColorf color_diffuse)
 {
 	m_camera_pos = camera_p;
-	m_color = color;
-	//m_brightness = brightness;
-	//dstream<<"m_brightness="<<m_brightness<<std::endl;
+	m_color.r = MYMIN(color_diffuse.r * m_color_bright.r + m_color_ambient.r, 1.0);
+	m_color.g = MYMIN(color_diffuse.g * m_color_bright.g + m_color_ambient.g, 1.0);
+	m_color.b = MYMIN(color_diffuse.b * m_color_bright.b + m_color_ambient.b, 1.0);
```

Comment:
Yes MIN is correct for limiting to 1.0.
I still think this should use the max of day colour and glow colour though, if i want red clouds at night but whte in daytime it's very difficult to work out what colour to use for day, because glow is always being added to it. So for each channel:
`m_color.r = MYMIN(MYMAX(color_diffuse.r * m_color_bright.r, m_color_ambient.r), 1.0);`